frontend/src/scenes/events/definitions/definitionDrawerLogic.ts

```diff
@@ -50,34 +105,52 @@ export const definitionDrawerLogic = kea<definitionDrawerLogicType<EventOrPropTy
                 closeDrawer: () => false,
             },
         ],
+        description: [
+            '',
+            {
+                setDescription: (_, { description }) => description,
+            },
+        ],
         definition: [
             null as EventOrPropType | null,
             {
-                setDefinition: (_, { definition }) => definition,
+                setDefinition: (state, { definition }) => {
+                    return { ...state, ...definition } as EventOrPropType
+                },
             },
         ],
-        description: [
-            '',
+        tagLoading: [
+            false,
             {
-                setDescription: (_, { description }) => description,
+                setTagLoading: (_, { loading }) => loading,
             },
         ],
         type: [
             '',
             {
-                setType: (_, { type }) => type,
+                setDrawerType: (_, { type }) => type,
             },
         ],
-        editing: [
-            false,
+        eventPropertiesDefinitions: [
+            [] as PropertyDefinition[],
             {
-                setDescriptionEditing: (_, { editing }) => editing,
+                setEventPropertyDefinition: (state, { propertyDefinition, id }) => {
+                    const newDefinitions = state.map((p) =>
+                        p.id === id ? Object.assign({}, { ...p, ...propertyDefinition }) : p
```

Comment:
`Object.assign({}, { ...p, ...propertyDefinition })` does exactly the same as just `{ ...p, ...propertyDefinition }`.